storage/pruning/pruningStorer.go

```diff
@@ -967,8 +991,10 @@ func (ps *PruningStorer) processPersistersToClose() []*persisterData {
 	}
 
 	if ps.numOfActivePersisters < uint32(len(ps.activePersisters)) {
-		persistersToClose = ps.activePersisters[ps.numOfActivePersisters:]
-		ps.activePersisters = ps.activePersisters[:ps.numOfActivePersisters]
+		numPersistersToKeep := ps.getNumPersistersToKeep(lastEpochNeeded)
+		allowedNumPersistersToKeep := core.MaxUint32(numPersistersToKeep, ps.numOfActivePersisters)
```

Comment:
For some implementations (eg: normal `pruningStorer`), `lastEpochNeeded` is always the newest epoch, because there are no other constraints. But `numOfActivePersisters` from config could demand more dbs to be kept opened, thus the need for `MaxUint32`.